TestCRDDeletionCascading flake

@kubernetes/sig-api-machinery-bugs 
/kind test-flake
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/59325/pull-kubernetes-unit/78868/

@jennybuckley gave a new failure sample: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/60331/pull-kubernetes-unit/83015/#k8siokubernetestestintegrationgarbagecollector-testcrddeletioncascading. (And the triage dashboard can't find it).

Anyway, the test [log](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/60331/pull-kubernetes-unit/83015/artifacts/junit_e9d37341e9f7ab19ac436e5365ba6ae7c20eea71_20180226-204814.stdout) showed similar behavior.

The tested started at I0226 20:57:02.82247.
The gc behaved correctly:
```
I0226 20:57:19.869254   26923 garbagecollector.go:466] delete object [v1/ConfigMap, namespace: crd-mixed, name: dependentxkkr2, uid: 9c8e8ae2-1b37-11e8-804b-0242ac110002] with propagation policy Background
```
And the 5min timeout occurred 23s after the test began.
```
I0226 20:57:25.773446   26923 garbagecollector.go:177] no resource updates from discovery, skipping garbage collector sync
panic: test timed out after 5m0s
```

I'll check what's the 5min timeout for.

--- a/hack/jenkins/test-dockerized.sh
+++ b/hack/jenkins/test-dockerized.sh
@@ -44,7 +44,6 @@ export KUBE_JUNIT_REPORT_DIR=${WORKSPACE}/artifacts
 export ARTIFACTS_DIR=${WORKSPACE}/artifacts
 # Save the verbose stdout as well.
 export KUBE_KEEP_VERBOSE_TEST_OUTPUT=y
-export KUBE_TIMEOUT='-timeout 300s'
 export KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=4
 export LOG_LEVEL=4
 

